javaslang-benchmark/src/test/java/javaslang/benchmark/collection/VectorBenchmark.java

```diff
@@ -11,80 +13,174 @@
 import static javaslang.benchmark.JmhRunner.*;
 
 public class VectorBenchmark {
+    @Test
+    public void testAsserts() throws Exception {
+        JmhRunner.runDebug(VectorBenchmark.class);
+    }
+
     public static void main(String... args) {
-        JmhRunner.runQuick(VectorBenchmark.class);
+        JmhRunner.runQuick(
+                Create.class,
+                Head.class,
+                Tail.class,
+                Get.class,
+                Update.class,
+                Prepend.class,
+                Append.class,
+                GroupBy.class,
+                Slice.class,
+                Iterate.class
+        );
     }
 
     @State(Scope.Benchmark)
     public static class Base {
-        @Param({ "10", "100", "1000" })
+        @Param({ "16", "500", "16000"/*, "500000", "16000000"*/ }) // i.e. depth 1,2,3(,4,5) for a branching factor of 32
```

Comment:
I'm currently working on optimizing vector by rewriting it to a `bit-mapped vector trie`.

Creation, getting elements at given indices and iteration is working.
`get` is currently `~2.5x` faster than `Clojure` and `~1.3x` faster than `Scala`. This will probably deteriorate once I extend it with `append`, `prepend` and `update`.

@danieldietrich, please assingn the corresponding issue to me.